packages/cojson/src/coValueCore/coValueCore.ts

```diff
@@ -610,27 +619,39 @@ export class CoValueCore {
 
     this.#batchedUpdates = false;
 
-    for (const listener of this.listeners) {
+    for (const [listener, listenerSettings] of this.listeners.entries()) {
       try {
-        listener(this, () => {
-          this.listeners.delete(listener);
-        });
+        listener(
+          this,
+          () => {
+            this.listeners.delete(listener);
+          },
+          listenerSettings.pendingDiffs,
+        );
+        listenerSettings.pendingDiffs = [];
       } catch (e) {
         logger.error("Error in listener for coValue " + this.id, { err: e });
       }
     }
   }
 
   subscribe(
-    listener: (core: CoValueCore, unsub: () => void) => void,
-    immediateInvoke = true,
+    listener: CoValueCoreCallback,
+    options?: { diffs?: boolean; skipInitial?: boolean },
   ): () => void {
-    this.listeners.add(listener);
+    this.listeners.set(listener, {
+      diffs: options?.diffs ?? false,
+      pendingDiffs: [],
+    });
 
-    if (immediateInvoke) {
-      listener(this, () => {
-        this.listeners.delete(listener);
-      });
+    if (!options?.skipInitial) {
+      listener(
+        this,
+        () => {
+          this.listeners.delete(listener);
+        },
+        "initial",
```

Comment:
this is just a marker - it's up to the CoValue view to translate the initial state into a corresponding diff (see CoMap implementation)